internal/config/config.go

```diff
@@ -947,3 +988,66 @@ func (c QEMUTDX) EqualTo(other AttestationCfg) (bool, error) {
 func toPtr[T any](v T) *T {
 	return &v
 }
+
+// auxAzureSEVSNP is a helper struct for unmarshaling the config from YAML for handling the version parsing.
+// The version fiels are kept to make it convertable to the native AzureSEVSNP struct.
+type auxAzureSEVSNP struct {
+	// description: |
+	//   Expected TPM measurements.
+	Measurements measurements.M `json:"measurements" yaml:"measurements" validate:"required,no_placeholders"`
+	// description: |
+	//   Lowest acceptable bootloader version.
+	BootloaderVersion uint8 `yaml:"-"`
+	// description: |
+	//   Lowest acceptable TEE version.
+	TEEVersion uint8 `json:"teeVersion" yaml:"-"`
+	// description: |
+	//   Lowest acceptable SEV-SNP version.
+	SNPVersion uint8 `json:"snpVersion" yaml:"-"`
+	// description: |
+	//   Lowest acceptable microcode version.
+	MicrocodeVersion uint8 `json:"microcodeVersion" yaml:"-"`
+	// description: |
+	//   Configuration for validating the firmware signature.
+	FirmwareSignerConfig SNPFirmwareSignerConfig `json:"firmwareSignerConfig" yaml:"firmwareSignerConfig"`
+	// description: |
+	//   AMD Root Key certificate used to verify the SEV-SNP certificate chain.
+	AMDRootKey Certificate `json:"amdRootKey" yaml:"amdRootKey"`
+}
+
+// fusedAzureSEVSNP is a helper struct for unmarshaling the config from YAML for handling the version parsing.
+type fusedAzureSEVSNP struct {
+	*auxAzureSEVSNP `yaml:",inline"`
+	// description: |
+	//   Lowest acceptable bootloader version.
+	BootloaderVersion string `yaml:"bootloaderVersion"`
+	// description: |
+	//   Lowest acceptable bootloader version.
+	TEEVersion string `yaml:"teeVersion"`
+	// description: |
+	//   Lowest acceptable bootloader version.
+	SNPVersion string `yaml:"snpVersion"`
+	// description: |
+	//   Lowest acceptable bootloader version.
+	MicrocodeVersion string `yaml:"microcodeVersion"`
+}
+
+func convertStringToUint(c *AzureSEVSNP, versionType version.Type, aux *fusedAzureSEVSNP) bool {
+	v, stringV := getUintAndStringPtrToVersion(versionType, c, aux)
+
+	bvInt, err := strconv.ParseInt(*stringV, 10, 8)
+	if err != nil {
+		return false
+	}
+	*v = uint8(bvInt)
+	return true
+}
```

Comment:
```suggestion
func convertStringToUint(c *AzureSEVSNP, versionType version.Type, aux *fusedAzureSEVSNP) error {
	v, stringV := getUintAndStringPtrToVersion(versionType, c, aux)

	bvInt, err := strconv.ParseInt(*stringV, 10, 8)
	if err != nil {
		return err
	}
	*v = uint8(bvInt)
	return nil
}
```
No need to lose information by not returning the error